### PR TITLE
chore(content-explorer): Migrate shareDialog

### PR DESCRIPTION
**Pull request by greg-in-a-box**
before:
image

after:
image

### CODE REVIEW

File: src/elements/content-explorer/ShareDialog.js.flow
Comment (tjuanitas) [inline src/elements/content-explorer/ShareDialog.js.flow:1]: do we need the flow files for the internal components of ContentExplorer? shouldn't the flow only be needed for the outer component that consumers use?

File: src/elements/content-explorer/ShareDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+                <div className="be-modal-input-group">
+                    <input
+                        ref={input => {
+                            textInput = input;
+                        }}
+                        onChange={noop}
+                        type="text"
+                        value={url}
+                    />
+                    <Button autoFocus className="be-modal-button-copy" onClick={copy} variant="primary">
```
Comment (tjuanitas) [inline src/elements/content-explorer/ShareDialog.tsx:75]: does button have `autoFocus` prop?

File: tsconfig.json [hunk 2]
```diff
@@ -15,5 +15,6 @@
         "strict": false
     },
     "files": [".storybook/typings.d.ts"],
-    "include": ["src/**/*.ts", "src/**/*.tsx", "examples/src/**.tsx", "types.ts", "scripts/jest/jest-setup.ts"]
+    "include": ["src/**/*.ts", "src/**/*.tsx", "examples/src/**.tsx", "types.ts", "scripts/jest/jest-setup.ts"],
+    "exclude": ["src/**/*.stories.ts", "src/**/*.stories.tsx"]
```
Comment (tjuanitas) [inline tsconfig.json:19]: why do we want to exclude stories?

File: src/elements/content-explorer/ShareDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+            contentLabel={formatMessage(messages.shareDialogLabel)}
+            isOpen={isOpen}
+            onRequestClose={onCancel}
+            overlayClassName={CLASS_MODAL_OVERLAY}
+            parentSelector={() => parentElement}
+            portalClassName={`${CLASS_MODAL} be-modal-share`}
+        >
+            <div className="be-modal-content">
+                <Text as="label">
+                    <FormattedMessage tagName="div" {...messages.shareDialogText} />
```
Comment (tjuanitas) [inline src/elements/content-explorer/ShareDialog.tsx:64]: do we need `tagName` still? i assume it was originally added for some block level style but i'm not sure if it's necessary

Comment (greg-in-a-box) [reply src/elements/content-explorer/ShareDialog.tsx:64]: going to remove this that should be there

File: tsconfig.json (hunk 2 above)
Comment (greg-in-a-box) [reply tsconfig.json:19]: it dawned on me that components and elements in stories do not need all the required props to demonstrate specific functions.

File: src/elements/content-explorer/ShareDialog.js.flow
Comment (greg-in-a-box) [reply src/elements/content-explorer/ShareDialog.js.flow:1]: do you think we should leave it out? its only there because i ran the codemod on it

**Review by jpan-box — APPROVED** [review]

**Review by tjuanitas — APPROVED** [review]

**Pull request merged**